base/ca/src/main/java/com/netscape/cms/servlet/cert/scep/CRSEnrollment.java

```diff
@@ -1284,6 +1314,27 @@ public void unwrapPKCS10(CRSPKIMessage req, CryptoContext cx)
 
     }
 
+    private SymmetricKey moveSymmetricToInternalToken(CryptoContext cx, SymmetricKey sk, SymmetricKey.Type skt, EncryptionAlgorithm ea)
+            throws Exception {
+        KeyPairGeneratorSpi.Usage[] usage = {
+                KeyPairGeneratorSpi.Usage.WRAP,
+                KeyPairGeneratorSpi.Usage.UNWRAP,
+                KeyPairGeneratorSpi.Usage.ENCRYPT,
+                KeyPairGeneratorSpi.Usage.DECRYPT};
+        KeyPair keyPairWrap = CryptoUtil.generateRSAKeyPair(cx.getInternalToken(), 2048, true, true, false, usage, usage);
```

Comment:
@fmarco76 , Shouldn't these temporary certs/keys need to be explicitly removed after each use or otherwise they would keep growing in the nssdb (please double check with @jmagne )?  Another way is to do this once and kept around for the lifetime of the system until it gets restarted.

Reply:
@ladycfu If I interpreted correctly the [documentation in JSS code](https://github.com/dogtagpki/jss/blob/master/base/src/main/java/org/mozilla/jss/crypto/KeyPairGenerator.java#L129-L138) temporary key pairs are not stored in the nssdb and deleted when the object containing them is cleaned by the garbage collector. I have also checked the list of key in the internaldb (the one in `/etc/pki/<instance_name>/alias`, but I am not sure it is the token used for this operation) using `certutil` and there are not extra keys after a SCEP enrollment. Anyway, I will wait confirmation from @jmagne because this key management is very tricky.

Reply:
@fmarco76 yes now that you mentioned it, that jogged my memory.   Also, I saw your other message to move to v10.13, so I will put my ack in that new PR. Thanks!